docs/authentication/jwt-auth.html

```diff
@@ -16,13 +16,407 @@
+    guard let authComponents = authHeader?.components(separatedBy: " "),
+        authComponents.count == 2,
+        authComponents[0] == "Bearer",
+        let jwt = try? JWT&lt;ClaimsStandardJWT&gt;(jwtString: authComponents[1], verifier: App.jwtVerifier)
+    else {
+        let _ = response.send(status: .unauthorized)
+        return try response.end()
+    }
+    response.send(jwt)
+    next()
+}</code></pre>
+    <div class="underline"></div>
+
+    <h2 class="heading-2"><span class="blue-text">Step 8:</span> Test the protected Route</h2>
+    <p>To test this, restart your server and send the POST request from Step 6.</p>
+    <p>Copy the returned JWT String and paste it into the following curl request:<p>
+<pre><code>curl -X GET \
+http://localhost:8080/jwtprotected \
+-H 'content-type: application/json' \
+-H 'Authorization: Bearer &lt;Your JWT String here&gt;'</code></pre>
+<p>You should see your JWT with your username returned to you. This should look something like:</p>
+<pre><code>{"claims":{"iss":"Kitura","sub":"Joe Bloggs","exp":574703307.61258602},"header":{"typ":"JWT","alg":"ES256"}}</code></pre>
+<p>Congratulations! You have just created a JWT single sign on system using a Kitura Server.
+     Your completed `JWTRoutes.swift` file for hs256 should look as follows:</p>
+<pre><code class="language-swift">import KituraContracts
+import SwiftJWT
+import Foundation
+
+
+func initializeJWTRoutes(app: App) {
+
+    app.router.post("/jwtlogin") { request, response, next in
+        let credentials = try request.read(as: UserCredentials.self)
+        let myClaims = ClaimsStandardJWT(iss: "Kitura", sub: credentials.username, exp: Date(timeIntervalSinceNow: 3600))
+        var myJWT = JWT(claims: myClaims)
+        let signedJWT = try myJWT.sign(using: App.jwtSigner)
+        response.send(signedJWT)
+        next()
+    }
+
+    app.router.get("/jwtprotected") { request, response, next in
+        let authHeader = request.headers["Authorization"]
+            guard let authComponents = authHeader?.components(separatedBy: " "),
+                authComponents.count == 2,
+                authComponents[0] == "Bearer",
+                let jwt = try? JWT&lt;ClaimsStandardJWT&gt;(jwtString: authComponents[1], verifier: App.jwtVerifier)
+                else {
+                    let _ = response.send(status: .unauthorized)
+                    return try response.end()
+            }
+        response.send(jwt)
+        next()
+    }
+}
+
+extension App {
+    // Example for HMAC signer and verifier
+    static let jwtSigner = JWTSigner.hs256(key: Data("kitura".utf8))
+    static let jwtVerifier = JWTVerifier.hs256(key: Data("kitura".utf8))
+}</code></pre>
+    <div class="underline"></div>
+    <h2 class="heading-2"><span class="blue-text">Step 9:</span> JWTs on Codable Routes (Optional)</h2>
+    <p class="block-text">In our example we used raw routing since we wanted to access the request headers.
```

Comment:
I think we should make this clearer... presumably we used raw routing because "we chose to pass the user credentials via the request headers"? In the second part of this we need to say something along the lines of "To use JWTs on codable routes, we need to encapsulate the code in a TypeSafeMiddleware, ...." and explain why we need to use a TypeSafeMiddleware as I'm not sure why we need to do this?